frontend/javascripts/oxalis/model/volumetracing/volumelayer.js

```diff
@@ -238,6 +239,130 @@ class VolumeLayer {
     return iterator;
   }
 
+  perpendicularVector2(pos1: Vector2, pos2: Vector2): Vector2 {
+    const dx = pos2[0] - pos1[0];
+    let perpendicularVector;
+    if (dx === 0) {
+      perpendicularVector = [1, 0];
+    } else {
+      const gradient = (pos2[1] - pos1[1]) / dx;
+      perpendicularVector = [gradient, -1];
+      const norm = this.normOfVector2(perpendicularVector);
+      perpendicularVector = this.scalarMultiplication(perpendicularVector, 1 / norm);
+    }
+    return perpendicularVector;
+  }
+
+  normOfVector2(vector: Vector2): number {
+    let norm = 0;
+    let i;
+    for (i of Vector2Indicies) {
```

Comment:
To be in line with the remainder of our code and to avoid `let` variables, I would replace this with 
```suggestion
    for (const i of Vector2Indicies) {
```
which is easier to reason about. Would be great if you could change this for alle the vector2 methods you created.